src/createSlice.ts

```diff
@@ -265,7 +350,86 @@ export function createSlice<
   return {
     name,
     reducer,
-    actions: actionCreators as any,
-    caseReducers: sliceCaseReducersByName as any
+    actions: context.actionCreators as any,
+    caseReducers: context.sliceCaseReducersByName as any
+  }
+}
+
+// --- internal functions & types ---
+
+interface ReducerHandlingContext<State> {
+  sliceCaseReducersByName: Record<string, CaseReducer<State, any>>
+  sliceCaseReducersByType: Record<string, CaseReducer<State, any>>
+  actionCreators: Record<string, Function>
+}
+
+interface ReducerDetails {
+  reducerName: string
+  type: string
+}
+
+function getReducerCreators<State>(): ReducerCreators<State> {
+  return {
+    asyncThunk(payloadCreator, config) {
```

Comment:
Something like 
```ts
  const slice = createSlice({
    name: 'test',
    initialState: {} as TestState,
    reducers: withSpecial => ({
      ...withSpecial.asyncThunks<RootState, Extra, Whatever>({
        thunk1Name: {payloadCreator, pending, rejected, fulfilled, options},
        thunk2Name: {payloadCreator, pending, rejected, fulfilled, options},
      }),
```
*could* potentially work, but type inference from the `payloadCreator` to the `pending`/`rejected`/`fulfilled` would be *very* flimsy - the `prepare` notation at the moment often requires manual type annotation and only checks that nothing wrong was annotated; we'd probably be running into similar problems and very complex types that way.

> Also, what's the need for having the payload arg defined separately?

The splitting up was necessary as otherwise in ` fulfilledReducer(state, action)`, `action.payload` would always infer to `unknown` and I didn't find another way around that.

Reply:
PS: also, the `Rejected` value might change within the thunks, making this even more difficult, as that can never be inferred.